gateway-server/src/main/java/org/apache/knox/gateway/util/KnoxCLI.java

```diff
@@ -652,34 +653,49 @@ public void execute() throws Exception {
          if (!ks.isCredentialStoreForClusterAvailable(GATEWAY_CREDENTIAL_STORE_NAME)) {
 //           log.creatingCredentialStoreForGateway();
            ks.createCredentialStoreForCluster(GATEWAY_CREDENTIAL_STORE_NAME);
-         }
-         else {
+         } else {
 //           log.credentialStoreForGatewayFoundNotCreating();
          }
          // LET'S NOT GENERATE A DIFFERENT KEY PASSPHRASE BY DEFAULT ANYMORE
          // IF A DEPLOYMENT WANTS TO CHANGE THE KEY PASSPHRASE TO MAKE IT MORE SECURE THEN
          // THEY CAN ADD THE ALIAS EXPLICITLY WITH THE CLI
          //as.generateAliasForCluster(GATEWAY_CREDENTIAL_STORE_NAME, GATEWAY_IDENTITY_PASSPHRASE);
        } catch (KeystoreServiceException e) {
-         throw new ServiceLifecycleException("Keystore was not loaded properly - the provided (or persisted) master secret may not match the password for the keystore.", e);
+         throw new ServiceLifecycleException("Keystore was not loaded properly - the stored password may not match the password for the keystore.", e);
        }
 
        try {
          if (!ks.isKeystoreForGatewayAvailable()) {
 //           log.creatingKeyStoreForGateway();
            ks.createKeystoreForGateway();
-         }
-         else {
+         } else {
 //           log.keyStoreForGatewayFoundNotCreating();
          }
-         char[] passphrase = as.getPasswordFromAliasForCluster(GATEWAY_CREDENTIAL_STORE_NAME, GATEWAY_IDENTITY_PASSPHRASE);
-         if (passphrase == null) {
-           MasterService ms = services.getService("MasterService");
-           passphrase = ms.getMasterSecret();
+         boolean isSelfSigned;
```

Comment:
Moving this declaration where you initiate the variable?

Reply:
fixing